notebooks/02-Distributed-Training_comments.ipynb

```diff
@@ -0,0 +1,962 @@
+    "                dtype=dtypes[key],\n",
+    "                device=get_device(),\n",
+    "            )\n",
+    "    return tensor_batch"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Sample batch\n",
+    "sample_batch = train_ds.take_batch(batch_size=3)\n",
+    "collate_fn(batch=sample_batch)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Model registry"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We'll be creating a model registry in our [Anyscale user storage](https://docs.anyscale.com/configuration/storage/#user-storage) to save our model checkpoints to. We'll be using OSS mlflow but we can easily [set up other experiment trackers](https://docs.ray.io/en/latest/train/user-guides/experiment-tracking.html) with Ray."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import shutil"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_registry = \"/mnt/user_storage/mlflow/doggos\"\n",
+    "os.path.isdir(model_registry) and shutil.rmtree(model_registry)  # clean up\n",
+    "os.makedirs(model_registry, exist_ok=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Training"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We'll define our training workload by specifying our:\n",
+    "- experiment and model parameters\n",
+    "- compute scaling configuration\n",
+    "- forward pass for batches of training and validation data\n",
+    "- train loop for each epoch of data (and checkpointing)\n",
+    "\n",
+    "<img src=\"https://raw.githubusercontent.com/anyscale/foundational-ray-app/refs/heads/main/images/trainer.png\" width=500>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Train loop config\n",
+    "experiment_name = \"doggos\"\n",
+    "train_loop_config = {\n",
+    "    \"model_registry\": model_registry,\n",
+    "    \"experiment_name\": experiment_name,\n",
+    "    \"embedding_dim\": 512,\n",
+    "    \"hidden_dim\": 256,\n",
+    "    \"dropout_p\": 0.3,\n",
+    "    \"lr\": 1e-3,\n",
+    "    \"lr_factor\": 0.8,\n",
+    "    \"lr_patience\": 3,\n",
+    "    \"num_epochs\": 20,\n",
+    "    \"batch_size\": 256,\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Scaling config\n",
+    "num_workers = 4\n",
+    "scaling_config = ray.train.ScalingConfig(\n",
+    "    num_workers=num_workers,\n",
+    "    use_gpu=True,\n",
+    "    resources_per_worker={\"CPU\": 8, \"GPU\": 1})"
```

Comment:
```suggestion
```

Setting resources_per_worker is an advanced configuration that's not needed here (it just steals logical CPUs from Ray Data).